src/ray/object_manager/plasma/object_store.cc

```diff
@@ -0,0 +1,172 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "ray/object_manager/plasma/object_store.h"
+
+namespace plasma {
+
+ObjectStore::ObjectStore(IAllocator &allocator)
+    : allocator_(allocator), object_table_() {}
+
+const LocalObject *ObjectStore::CreateObject(const ray::ObjectInfo &object_info,
+                                             plasma::flatbuf::ObjectSource source,
+                                             bool fallback_allocate) {
+  RAY_LOG(DEBUG) << "attempting to create object " << object_info.object_id << " size "
+                 << object_info.data_size;
+  RAY_CHECK(object_table_.count(object_info.object_id) == 0)
+      << object_info.object_id << " already exists!";
+  auto object_size = object_info.GetObjectSize();
+  auto allocation = fallback_allocate ? allocator_.FallbackAllocate(object_size)
+                                      : allocator_.Allocate(object_size);
+  if (!allocation.has_value()) {
+    return nullptr;
+  }
+  auto ptr = std::make_unique<LocalObject>(std::move(allocation.value()));
+  auto entry =
+      object_table_.emplace(object_info.object_id, std::move(ptr)).first->second.get();
+  entry->object_info = object_info;
+  entry->state = ObjectState::PLASMA_CREATED;
+  entry->create_time = std::time(nullptr);
+  entry->construct_duration = -1;
+  entry->source = source;
+
+  num_objects_unsealed_++;
+  num_bytes_unsealed_ += entry->GetObjectSize();
+  num_bytes_created_total_ += entry->GetObjectSize();
+  RAY_LOG(DEBUG) << "create object " << object_info.object_id << " succeeded";
```

Comment:
Add "create object failed" in line 36 so that we will log the failure case as well?